erpnext/agriculture/doctype/harvest/harvest.py

```diff
@@ -3,8 +3,52 @@
 # For license information, please see license.txt
 
 from __future__ import unicode_literals
-# import frappe
+import frappe
+import json
+from frappe import _
 from frappe.model.document import Document
 
 class Harvest(Document):
 	pass
+
+@frappe.whitelist()
+def create_stock_entry(harvest):
+	harvest = frappe.get_doc("Harvest", harvest)
+	if stock_entry_exists(harvest.get('name')):
+		stock_entry_name = frappe.db.get_value("Stock Entry", {"harvest": harvest.get('name')}, fieldname = ['name'])
+		return frappe.msgprint(_('Stock Entry {0} has been already created against this harvest.').format(frappe.utils.get_link_to_form("Stock Entry", stock_entry_name)))
+	stock_entry = frappe.new_doc('Stock Entry')
+	stock_entry.harvest = harvest.get('name')
+	stock_entry.purpose = harvest.get('purpose')
+	stock_entry.stock_entry_type = 'Harvest'
+
+	if harvest.get('strain'):
+		stock_entry = update_stock_entry_based_on_strain(harvest, stock_entry)
+
+	stock_entry.set_incoming_rate()
```

Comment:
The system let me create a Strain without any produced items or byproducts, and it failed when I tried to create a Stock Entry from a harvest from that strain.

```python
Traceback (most recent call last):
  File "/home/rohan/psych/apps/frappe/frappe/app.py", line 64, in application
    response = frappe.api.handle()
  File "/home/rohan/psych/apps/frappe/frappe/api.py", line 59, in handle
    return frappe.handler.handle()
  File "/home/rohan/psych/apps/frappe/frappe/handler.py", line 24, in handle
    data = execute_cmd(cmd)
  File "/home/rohan/psych/apps/frappe/frappe/handler.py", line 63, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "/home/rohan/psych/apps/frappe/frappe/__init__.py", line 1054, in call
    return fn(*args, **newargs)
  File "/home/rohan/psych/apps/erpnext/erpnext/agriculture/doctype/harvest/harvest.py", line 28, in create_stock_entry
    stock_entry.set_incoming_rate()
  File "/home/rohan/psych/apps/erpnext/erpnext/stock/doctype/stock_entry/stock_entry.py", line 374, in set_incoming_rate
    for d in self.items:
AttributeError: 'StockEntry' object has no attribute 'items'
```

Reply:
check added for is child table exist then call these functions